### DISCUSSION TITLE
Fluent API for Rector\Config\RectorConfig

### DISCUSSION THREAD

**Discussion by gisostallenberg** [rectorphp/rector#7138] [Ideas]
I love the new config which adds autocomplete!

IMO it would be nice if the methods return the instance so the configuration can be done in a fluent style.
e.g.:
```php
$rectorConfig
    ->importNames()
    ->phpVersion(PhpVersion::PHP_81);
```
instead of
```php
$rectorConfig->importNames();
$rectorConfig->phpVersion(PhpVersion::PHP_81);
```

What do you think?

**Comment by TomasVotruba** [comment]
Hi, thank you :)

To be honest, I'm not very fan of fluent interfaces, as they introduce 3 options to use an object. With new, with fluent, with object assign.

* https://www.yegor256.com/2018/03/13/fluent-interfaces.html
* https://ocramius.github.io/blog/fluent-interfaces-are-evil

It also look like the import and php versions are somehow related and depend on each other, which is not true.
It's hell to maintain or decouple from and PHPStan and Rector have troubles to analyse it.
It makes sense only in case of some query builder, where combination of methods are required to call. Not just because the object is the same.
